scrapy/http/request/form.py

```diff
@@ -107,8 +108,13 @@ def _get_inputs(form, formdata, dont_click, clickdata, response):
 
     inputs = form.xpath('descendant::textarea'
                         '|descendant::select'
-                        '|descendant::input[@type!="submit" and @type!="image" and @type!="reset"'
-                        'and ((@type!="checkbox" and @type!="radio") or @checked)]')
+                        '|descendant::input[not(@type) or @type['
+                        ' translate(., "SUBMIT", "submit") != "submit"'
```

Comment:
_" there should be a performance penalty"_
probably. I'd be curious to see a comparison.

and I'd still go for the regexp version for readability